mycore-base/src/main/java/org/mycore/frontend/cli/MCRBasicCommands.java

```diff
@@ -219,6 +233,74 @@ private static void createSampleConfigFile(String path) throws IOException {
+                    .getContent(Filters.element("mapping-file", nsPersistence));
+                List<String> oldMappings = mappingElements.stream()
+                    .map(x -> x.getTextNormalize())
+                    .collect(Collectors.toList());
+
+                mappingElements
+                    .removeIf(e -> e.getTextNormalize().endsWith("-mappings.xml")
+                        && MCRBasicCommands.class.getResource(
+                            (e.getTextNormalize().startsWith("/") ? "" : "/") + e.getTextNormalize()) == null);
+
+                if (oldMappings.size() > mappingElements.size()) {
+                    LOGGER.warn(Integer.toString(oldMappings.size() - mappingElements.size())
+                        + " unknown mapping files removed from '" + persistenceXMLFile.toString() + "'.");
+                }
+
+                List<String> newMappings = new ArrayList<String>();
+                for (MCRComponent cmp : MCRRuntimeComponentDetector.getAllComponents()) {
+                    try (ZipInputStream zip = new ZipInputStream(Files.newInputStream(cmp.getJarFile().toPath()))) {
+                        ZipEntry ze = null;
+                        while ((ze = zip.getNextEntry()) != null) {
+                            String zeName = ze.getName();
+                            if (zeName.startsWith("META-INF/") && zeName.endsWith("-mappings.xml")
+                                && !oldMappings.contains(zeName)) {
+                                newMappings.add(zeName);
+                            }
+                        }
+                    }
+                }
+                if (newMappings.size() > 0) {
+                    Comment c = new Comment(
+                        " mapping files, added by command 'reload mappings in jpa configuration file' ");
+                    ePersistenceUnit.getContent(Filters.comment()).removeIf(x -> x.getText().equals(c.getText()));
+                    ePersistenceUnit.addContent(0, c);
+
+                    int pos = 0;
+                    for (String mappingFile : newMappings) {
+                        Element eMappingFile = new Element("mapping-file", nsPersistence).setText(mappingFile);
+                        ePersistenceUnit.addContent(++pos, eMappingFile);
+                    }
+                    LOGGER
+                        .warn(newMappings.size() + " mapping files added to '" + persistenceXMLFile.toString() + "'.");
+                }
+
+                XMLOutputter out = new XMLOutputter(Format.getPrettyFormat());
+                try (BufferedWriter bw = Files.newBufferedWriter(persistenceXMLFile.toPath())) {
+                    out.output(doc, bw);
+                }
+
+            } else {
+                LOGGER.warn("The config file '" + persistenceXMLFile.toString() + "' does not exist yet!");
+            }
+        } catch (Exception e) {
+            LOGGER.error(e.getMessage(), e);
+        }
```

Comment:
Do not catch and log exception, throw it! We need the exception to detect if it ran successfully or not.

Reply:
rethrown as MCRException